src/main/java/it/pn/frontend/e2e/section/mittente/DettaglioNotificaMittenteSection.java

```diff
@@ -143,113 +144,108 @@ public void siVisualizzaPercosoNotifica() {
-            if (testoDaControllare.equals("Oggi")) {
-                String data = LocalDate.now().toString();
-                String[] date = data.split("-");
-                testoDaControllare = date[2] + "/" + date[1] + "/" + date[0];
+            String testoFile = pdfStripper.getText(pdfFile).replaceAll ("\r\n|\r|\n", " ");
+            if (testoDaControllare.equals("Oggi")){
+              String data =  LocalDate.now().toString();
+              String[] date = data.split("-");
+              testoDaControllare = date[2]+"/"+date[1]+"/"+date[0] ;
             }
-            if (testoFile.contains(testoDaControllare)) {
+            if (testoFile.contains(testoDaControllare)){
                 pdfFile.close();
                 return true;
             }
             pdfFile.close();
         } catch (IOException e) {
-            logger.error("File non trovato con errore: " + e.getMessage());
-            Assert.fail("File non trovato con errore: " + e.getMessage());
+            logger.error("File non trovato con errore: "+e.getMessage());
+            Assert.fail("File non trovato con errore: "+e.getMessage());
         }
         return false;
     }
 
     public void clickLinkDocumentiAllegati() {
-        if (this.linkAllegati.get(0).isDisplayed()) {
+        if (this.linkAllegati.get(0).isDisplayed()){
             this.linkAllegati.get(0).click();
-        } else {
+        }else {
             this.js().executeScript("arguments[0].scrollIntoView(true);", this.linkAllegati.get(0));
+            logger.info("click sul link allegati");
             this.linkAllegati.get(0).click();
         }
-
     }
 
     public void clickLinkAvvenutaRicezione(int i) {
-        if (this.linkAllegati.get(i).isDisplayed()) {
+        logger.info("click sul link avvenuta ricezione");
+        if (this.linkAllegati.get(i).isDisplayed()){
             this.linkAllegati.get(i).click();
-        } else {
-            this.js().executeScript("arguments[0].scrollIntoView(true);", this.linkAllegati.get(i));
+        }else {
+            this.js().executeScript("arguments[0].scrollIntoView(true);",this.linkAllegati.get(i));
             this.linkAllegati.get(i).click();
         }
-    }
+}
 
     public int getLinkAvvenutaRicezione() {
         return linkAllegati.size();
     }
 
     public void clickLinkAttestazioneOpponibile(String nomeFile) {
         try {
-            By fileLinkBy = By.cssSelector("[data-testid='download-legalfact']");
+            By fileLinkBy = By.xpath("//button[contains(text(),'"+nomeFile+"')]");
             this.getWebDriverWait(30).until(ExpectedConditions.visibilityOfElementLocated(fileLinkBy));
             List<WebElement> fileLink = this.elements(fileLinkBy);
-            for (WebElement fileName : fileLink) {
-                if (fileName.getText().contains(nomeFile)) {
-                    fileName.click();
-                    return;
-                }
+            if (fileLink.get(0).isDisplayed()){
```

Comment:
@leleOFA why this for loop has been updated with access to the first element of the list?

Reply:
it is the pull with development, I didn't change it